src/app/components/SelectNode/UploadMacaroons.tsx

```diff
@@ -1,171 +1,216 @@
+      if (this.props.onComplete) {
+        this.props.onComplete();
+      }
+    }
     if (prevProps.error !== this.props.error) {
-      this.setState({ error: this.props.error });
+      this.setState({error: this.props.error});
     }
   }
 
   render() {
-    const { nodeType } = this.props;
-    const { error, admin, readonly, isShowingHexInputs } = this.state;
+    const {checkAuthError, nodeType} = this.props;
+    const {error, admin, readonly, isShowingHexInputs} = this.state;
     const dirs = (nodeType && DEFAULT_LND_DIRS[nodeType]) || DEFAULT_LND_DIRS[NODE_TYPE.LOCAL];
     return (
-      <Form layout="vertical" className="UploadMacaroons">
-        <div className="UploadMacaroons-description">
-          We need to authenticate with your node using macaroons. Your admin
-          macaroon will be encrypted, and payments will <em>never</em> be made
-          without your explicit approval.
-        </div>
-
-        {isShowingHexInputs ? (
-          <>
-            <Form.Item label="Admin macaroon">
-              <Input
-                name="admin"
-                value={admin}
-                onChange={this.handleChangeMacaroonHex}
-                placeholder="Paste hex string here"
-              />
-            </Form.Item>
-            <Form.Item label="Readonly macaroon">
-              <Input
-                name="readonly"
-                value={readonly}
-                onChange={this.handleChangeMacaroonHex}
-                placeholder="Paste hex string here"
-              />
-            </Form.Item>
-          </>
-        ) : (
-          <>
-            <Upload.Dragger
-              accept=".macaroon"
-              showUploadList={false}
-              beforeUpload={
-                (file) => this.handleMacaroonUpload('admin', file)
-              }
-            >
-              <p className="ant-upload-drag-icon">
-                <Icon type={admin ? 'check-circle' : 'inbox'} />
-              </p>
-              <p className="ant-upload-text">
-                Upload <code>admin.macaroon</code>
-              </p>
-              <p className="ant-upload-hint">
-                Click or drag to upload macaroon
-              </p>
-            </Upload.Dragger>
-
-            <Upload.Dragger
-              accept=".macaroon"
-              showUploadList={false}
-              beforeUpload={
-                (file) => this.handleMacaroonUpload('readonly', file)
-              }
-            >
-              <p className="ant-upload-drag-icon">
-                <Icon type={readonly ? 'check-circle' : 'inbox'} />
-              </p>
-              <p className="ant-upload-text">
-                Upload <code>readonly.macaroon</code>
-              </p>
-              <p className="ant-upload-hint">
-                Click or drag to upload
-              </p>
-            </Upload.Dragger>
-
-            {dirs &&
+      <div>
+        <TitleTemplate title={'Upload Macaroons'}/>
```

Comment:
Unfortunately the back button here doesn't work with LightningApp since it auto-connects. Might be good to have the back button here take you back to the selection of node type, rather than whatever the previous page was.